sources/buildengine/Xenko.Core.BuildEngine.Common/Command.cs

```diff
@@ -15,7 +15,7 @@
 
 namespace Xenko.Core.BuildEngine
 {
-    [DataContract(Inherited = true)]
+    [DataContract(Inherited = true), Serializable]
```

Comment:
I would prefer to use only our serialization (which is much more tested and already used in many places) rather than having a third way (`BinarySerialization` and Yaml) to serialize our type that will likely almost never be tested and might behave differently for various specific case (i.e. we use attribute on properties to express a bunch of stuff).

Is there a way, like WCF, to use a custom serialization engine?
After a quick check, it looks like it's not possible easily (https://github.com/tylerjensen/ServiceWire/wiki/Fast-Serialization-Overview mentions you have to use byte array if you want to avoid `BinaryFormatter`...)

Assuming we have to stick with `Serializable` still, did you test if this part work?
As far as I remember, building in separate process is used only for FBX so far (because their API is not threadsafe), so testing with a project having FBX is necessary if you didn't do it so far.

Reply:
Took a closer look at ServiceWire, it looks like we can do the same by implementing a custom `ServiceWire.ISerializer`  and passing it as parameter to `NpHost` ctor.
You can look at WCF serialization implementation `XenkoXmlObjectSerializer.EncodeObject` and `XenkoXmlObjectSerializer.DecodeObject`, the new `ServiceWire.ISerializer.Serialize` and `Deserialize` should be very close.

Reply:
Okay I'll give that a try. Yes i originally was using JSON serialization but decided to drop it to remove dependencies. I'll do the xml serialize instead.

Reply:
Just to be clear, it won't use XML at all (the class name might have misled you: WCF is using XML to communicate but Xenko generates `byte[]`, so we were wrapping our `byte[]` and wrapped it in XML data using base64 encoding).
If you check methods of last message, you will see that `ServiceWire.ISerializer` expects `byte[]` and the method to use as reference (`XenkoXmlObjectSerializer.EncodeObject` and `DecodeObject` are also working with `byte[]`.
Hopefully it should be trivial, let me know if you need any help.